### PR TITLE
Fix off-by-one error in span batching logic

### PR DESCRIPTION
**Pull request by richvdh**
(also, add a couple of type specifications that were helpful when I was
debugging).

The problem here was that we waited until we had 21 spans ready to send, then
sent them out in batches of 20, which basically meant we always ended up with a
batch of 20 and another batch with a single item.

### CODE REVIEW

**Comment by richvdh** [comment]
I'm actually a bit confused about what this fancy stuff with `queue` is for. I see that we add things to it at line 128, and then once it has 20 items, we atomically replace it with another queue at line 144, before dividing the queue into 20-item chunks. But as far as I can tell those things are happening in the same thread, so there is no way for us to end up with more than 20 items in the queue. Why don't we just turn the whole queue into a `jaeger::Batch`?

File: src/lib.rs [hunk 1]
```diff
@@ -138,10 +138,10 @@ impl Reporter {
 
                     // We batch up the spans before sending them, waiting at
                     // most N seconds between sends
-                    if queue.len() > 20 || (!queue.is_empty() && last_push.elapsed().as_secs() > 20)
+                    if queue.len() >= 20 || (!queue.is_empty() && last_push.elapsed().as_secs() > 20)
```
Comment (richvdh) [inline src/lib.rs:141]: this is the critical change here.

**Comment by erikjohnston** [comment]
> I'm actually a bit confused about what this fancy stuff with `queue` is for. I see that we add things to it at line 128, and then once it has 20 items, we atomically replace it with another queue at line 144, before dividing the queue into 20-item chunks. But as far as I can tell those things are happening in the same thread, so there is no way for us to end up with more than 20 items in the queue. 

The batching code has gone through a bunch of iterations and me generally faffing about, is basically the short answer.

> Why don't we just turn the whole queue into a `jaeger::Batch`?

We could, I have a vague preference for the cleanliness of having a `Vec` and then giving that to the `Batch` (which is basically free perf wise), but either works

**Comment by erikjohnston** [comment]
> (also, add a couple of type specifications that were helpful when I was
> debugging).

(FWIW plugins like rust-analyzer automatically annotate the types of variables, which is a) very useful and b) probably why the community tends not to be particularly explicit with types.)

**Pull request merged**